types/abci.pb.go

```diff
@@ -100,7 +100,7 @@ var xxx_messageInfo_TxResponse proto.InternalMessageInfo
 
 // ABCIMessageLog defines a structure containing an indexed tx ABCI message log.
 type ABCIMessageLog struct {
-	MsgIndex uint32 `protobuf:"varint,1,opt,name=msg_index,json=msgIndex,proto3" json:"msg_index,omitempty"`
+	MsgIndex uint32 `protobuf:"varint,1,opt,name=msg_index,json=msgIndex,proto3" json:"msg_index"`
```

Comment:
This file gets auto-generated, so we shouldn't modify it manually.

Could you modify the relevant `.proto` file, and run `make proto-gen`?

Reply:
That SO thread is for golang protobuf v1, and we gogoproto (for now). You can maybe try `[(gogoproto.jsontag) = "msg_index"]`.